app/controllers/library_hours_controller.rb

```diff
@@ -1,10 +1,43 @@
 # frozen_string_literal: true
 
 class LibraryHoursController < ApplicationController
-  before_action :get_locations, :set_dates, only: [:index]
+  before_action :buildings, :set_dates, only: [:index, :show]
 
   def index
+    @buildings.each do |building|
+      building[:spaces].map! do |space|
+        space = [building[:slug], LibraryHour.where(location_id: space, date: @monday..@sunday + 1)]
+      end
+    end
+  end
+
+  def set_dates
+    @today = Date.today
+    @date = params[:date].nil? ? @today : Date.parse(params[:date])
+
+    unless params[:date].nil?
+      @monday = @date.beginning_of_week
+      @sunday = @date.end_of_week
+      @next_week = @date.next_week
+      @last_week = @date.prev_week
+    else
+      @monday = @today.beginning_of_week
+      @sunday = @today.end_of_week
+      @next_week = @today.next_week
+      @last_week = @today.prev_week
+    end
+  end
+
+  def buildings
     @buildings = [
```

Comment:
@cdoyle-temple The `@buildings` is really a configuration and should be pulled in from a locale specific file kept in the `config` directory.  Let's talk about this one.

Reply:
This looks like a setter too, but since it's setting attributes of a building, it probably belongs in a concern.

Reply:
I've created a configuration in the applications config.  These are also controller instance variables, so I think we're good here too.